draft-yasskin-webpackage-use-cases.md

```diff
@@ -0,0 +1,365 @@
+arbitrary resources without a linear scan.
+
+## Streamed loading {#streamed-loading}
+
+The browser can load a package as it downloads.
+
+## Resources from multiple origins in a package {#multiple-origins}
+
+A package from origin `A` can contain resources from origin `B`
+authenticated at the same level as those from `A`.
+
+## Cryptographic agility {#crypto-agility}
+
+Obsolete cryptographic algorithms can be replaced.
+
+## Cross-signatures {#cross-signatures}
+
+Third-parties can vouch for packages by signing them.
+
+## Unsigned content {#unsigned-content}
+
+Alex can create their own package without a CA-signed
+certificate, and Bailey can view the content of the package.
+
+## Certificate revocation {#revocation}
+
+When a package is signed by a revoked certificate, online browsers
+can detect this reasonably quickly.
+
+## Downgrade prevention {#no-downgrades}
+
+Attackers can't cause a browser to trust an older, vulnerable
+version of a package after the browser has seen a newer version.
+
+# Nice to have
+
+## Binary {#binary}
+
+The format is identified as binary by tools that might try to "fix"
+line endings.
+
+## Deduplication of diamond dependencies {#deduplication}
+
+Nested packages that have multiple dependency routes to the same
+sub-package, can be transmitted and stored with only one copy of that
+sub-package.
+
+## Old crypto can be removed {#crypto-removal}
+
+The ecosystem can identify when an obsolete cryptographic algorithm is
+no longer needed and can be removed.
+
+## Compress transfers {#transfer-compression}
+
+Transferring a package over the network takes as few bytes as
+possible.
+
+## Compress stored packages {#stored-compression}
+
+Storing a package on disk takes as few bytes as possible.
+
+## Subsetting and reordering {#subsetting}
+
+Resources can be removed from and reordered within a package, without
+breaking [signatures](#signing).
+
+## Packaged validity information {#packaged-validity}
+
+{{revocation}}{:format="title"} and {{no-downgrades}}{:format="title"}
+information can itself be packaged or included in other packages.
+
+## Signing uses existing TLS certificates {#existing-certs}
+
+A "normal" TLS certificate can be used for signing packages. Avoiding
+extra requirements like "code signing" certificates makes packaging
+more accessible to all sites.
+
+## External dependencies {#external-dependencies}
+
+Sub-packages can be "external" to the main package, meaning the browser
+will need to either fetch them separately or already have them.
+([#35, App Installer Story](https://github.com/WICG/webpackage/issues/35))
+
+## Trailing length {#trailing-length}
+
+The package's length in bytes appears a fixed offset from the end of
+the package.
+
+# Non-goals
+
+Some features often come along with packaging and signing, and it's
+important to explicitly note that they don't appear in the list of
+{{requirements}}{:format="title"}.
+
+## Store confidential data {#confidential}
```

Comment:
Is this worth a little more enumeration?  Confidentiality comes up repeatedly in discussions and it'd be good to ensure that rehashing is minimized.